mypy/typeanal.py

```diff
@@ -1241,8 +1240,15 @@ def set_allow_param_spec_literals(self, to: bool) -> Iterator[None]:
 
 TypeVarLikeList = List[Tuple[str, TypeVarLikeExpr]]
 
-# Mypyc doesn't support callback protocols yet.
-MsgCallback = Callable[[str, Context, DefaultNamedArg(Optional[ErrorCode], 'code')], None]
+
+class MsgCallback(Protocol):
+    def __call__(
+        self,
+        __msg: str,
+        __ctx: Context,
+        *,
+        code: Optional[ErrorCode] = ...
```

Comment:
```suggestion
        code: Optional[ErrorCode] = None
```
Looks like this will make mypyc happy